test/controllers/benefits/CompanyBenefitControllerSpec.scala

```diff
@@ -194,6 +210,64 @@ class CompanyBenefitControllerSpec
       }
     }
 
+    "redirect to the Tax Account Summary Page (start of journey)" when {
+      "the form has the value noIDontGetThisBenefit and EndCompanyBenefit_BenefitTypeKey is not cached" in {
+        val SUT = createSUT
+        when(
+          journeyCacheService.mandatoryJourneyValueAs(eqTo(EndCompanyBenefit_BenefitTypeKey), any())(
+            any()))
+          .thenReturn(Future.successful(Left("")))
+
+        val result = SUT.submitDecision(
+          RequestBuilder
+            .buildFakeRequestWithAuth("POST")
+            .withFormUrlEncodedBody(DecisionChoice -> NoIDontGetThisBenefit))
+
+        val redirectUrl = redirectLocation(result)
+
+        redirectUrl mustBe Some(controllers.routes.TaxAccountSummaryController.onPageLoad().url)
+
+      }
+
+      "the form has the value YesIGetThisBenefit and EndCompanyBenefit_BenefitTypeKey is not cached" in {
+        val SUT = createSUT
+        when(
+          journeyCacheService.mandatoryJourneyValueAs(eqTo(EndCompanyBenefit_BenefitTypeKey), any())(
+            any()))
+          .thenReturn(Future.successful(Left("")))
+
+        val result = SUT.submitDecision(
+          RequestBuilder
+            .buildFakeRequestWithAuth("POST")
+            .withFormUrlEncodedBody(DecisionChoice -> YesIGetThisBenefit))
+
+        val redirectUrl = redirectLocation(result).getOrElse("")
+
+        redirectUrl mustBe controllers.routes.TaxAccountSummaryController.onPageLoad()
+
+      }
+
+      "the form has no valid value" in{
+        val SUT = createSUT
+        val benefitType = "Telephone"
+        when(
+          journeyCacheService.mandatoryJourneyValueAs[String](eqTo(EndCompanyBenefit_BenefitTypeKey), any())(
+            any()))
+          .thenReturn(Future.successful(Right(benefitType)))
+
```

Comment:
You can deduplicate this block into a function within this `should` scope

`val benefitType = "Telephone"
        when(
          journeyCacheService.mandatoryJourneyValueAs[String](eqTo(EndCompanyBenefit_BenefitTypeKey), any())(
            any()))
          .thenReturn(Future.successful(Right(benefitType)))`

Reply:
I've done this now for ensuring it exists, and ensuring it is clear (complete with the use of Telephone.name)